android-core/src/main/java/com/mparticle/internal/MParticleApiClientImpl.java

```diff
@@ -203,28 +206,27 @@ public void fetchConfig(boolean force) throws IOException, MPConfigException {
         }
     }
 
-    public JSONObject fetchAudiences() {
-
-        JSONObject response = null;
+    public void fetchUserAudience(BaseAudienceTask task){
+        JSONObject jsonResponse = null;
         try {
-            Logger.debug("Starting Segment Network request");
             MPConnection connection = getUrl(Endpoint.AUDIENCE).openConnection();
-            connection.setConnectTimeout(mConfigManager.getConnectionTimeout());
-            connection.setReadTimeout(mConfigManager.getConnectionTimeout());
             connection.setRequestProperty("User-Agent", mUserAgent);
-
             addMessageSignature(connection, null);
+            connection.setRequestProperty("x-mp-key", mApiKey);
             makeUrlRequest(Endpoint.AUDIENCE, connection, true);
             if (connection.getResponseCode() == HttpURLConnection.HTTP_FORBIDDEN) {
                 Logger.error("Segment call forbidden: is Segmentation enabled for your account?");
             }
-            response = MPUtility.getJsonResponse(connection);
-            parseCookies(response);
-
+            jsonResponse = MPUtility.getJsonResponse(connection);
```

Comment:
check identity but we should be logging:

- the request (presumably the full url with query params)
- the response status code
- the response body